src/System.Runtime/src/System/HashCode.cs

```diff
@@ -0,0 +1,250 @@
+        [TargetedPatchingOptOut("Performance critical for inlining across NGen images.")]
+        public static int Combine<T1, T2>(T1 value1, T2 value2)
+        {
+            var hc = new HashCode();
+            hc.Add(value1);
+            hc.Add(value2);
+            return hc.ToHashCode();
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        [TargetedPatchingOptOut("Performance critical for inlining across NGen images.")]
+        public static int Combine<T1, T2, T3>(T1 value1, T2 value2, T3 value3)
+        {
+            var hc = new HashCode();
+            hc.Add(value1);
+            hc.Add(value2);
+            hc.Add(value3);
+            return hc.ToHashCode();
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        [TargetedPatchingOptOut("Performance critical for inlining across NGen images.")]
+        public static int Combine<T1, T2, T3, T4>(T1 value1, T2 value2, T3 value3, T4 value4)
+        {
+            var hc = new HashCode();
+            hc.Add(value1);
+            hc.Add(value2);
+            hc.Add(value3);
+            hc.Add(value4);
+            return hc.ToHashCode();
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        [TargetedPatchingOptOut("Performance critical for inlining across NGen images.")]
+        public static int Combine<T1, T2, T3, T4, T5>(T1 value1, T2 value2, T3 value3, T4 value4, T5 value5)
+        {
+            var hc = new HashCode();
+            hc.Add(value1);
+            hc.Add(value2);
+            hc.Add(value3);
+            hc.Add(value4);
+            hc.Add(value5);
+            return hc.ToHashCode();
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        [TargetedPatchingOptOut("Performance critical for inlining across NGen images.")]
+        public static int Combine<T1, T2, T3, T4, T5, T6>(T1 value1, T2 value2, T3 value3, T4 value4, T5 value5, T6 value6)
+        {
+            var hc = new HashCode();
+            hc.Add(value1);
+            hc.Add(value2);
+            hc.Add(value3);
+            hc.Add(value4);
+            hc.Add(value5);
+            hc.Add(value6);
+            return hc.ToHashCode();
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        [TargetedPatchingOptOut("Performance critical for inlining across NGen images.")]
+        public static int Combine<T1, T2, T3, T4, T5, T6, T7>(T1 value1, T2 value2, T3 value3, T4 value4, T5 value5, T6 value6, T7 value7)
+        {
+            var hc = new HashCode();
+            hc.Add(value1);
+            hc.Add(value2);
+            hc.Add(value3);
+            hc.Add(value4);
+            hc.Add(value5);
+            hc.Add(value6);
+            hc.Add(value7);
+            return hc.ToHashCode();
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        [TargetedPatchingOptOut("Performance critical for inlining across NGen images.")]
```

Comment:
@jkotas is "targetedpatchingoptout" still needed in corfx?

Reply:
if so when do we apply it...

Reply:
> @jkotas is "targetedpatchingoptout" still needed in corfx?

No. These attributes can be deleted.